File: slimadapter/src/main/java/com/jhj/slimadapter/listener/OnItemDragListener.java
package com.jhj.slimadapter.listener;

import android.support.v7.widget.RecyclerView;

/**
 * 拖拽事件
 * Created by jhj on 18-10-11.
 */
public interface OnItemDragListener {

    /**
     * 可以设置开始动画
     *
     * @param viewHolder
     * @param pos
     */
    void onItemDragStart(RecyclerView.ViewHolder viewHolder, int pos);


    void onItemDragMoving(RecyclerView.ViewHolder source, int from, RecyclerView.ViewHolder target, int to);

    /**
     * 设置结束动画
     *
     * @param viewHolder
     * @param pos
     */
    void onItemDragEnd(RecyclerView.ViewHolder viewHolder, int pos);
}

File: slimadapterkt/src/main/java/com/zgdj/slimadapterkt/listener/OnItemDragListener.kt
package com.zgdj.slimadapterkt.listener

import android.support.v7.widget.RecyclerView

/**
 * 拖拽事件
 * Created by jhj on 18-10-11.
 */
interface OnItemDragListener {

    /**
     * 可以设置开始动画
     *
     * @param viewHolder
     * @param pos
     */
    fun onItemDragStart(viewHolder: RecyclerView.ViewHolder, pos: Int)


    fun onItemDragMoving(source: RecyclerView.ViewHolder, from: Int, target: RecyclerView.ViewHolder, to: Int)

    /**
     * 设置结束动画
     *
     * @param viewHolder
     * @param pos
     */
    fun onItemDragEnd(viewHolder: RecyclerView.ViewHolder, pos: Int)
}
